### PR TITLE
feat: introduce Maven configuration for GraalVM testing

### PR DESCRIPTION
**Pull request by dzou**
This PR introduces Maven project and Kokoro configuration for GraalVM native image testing.

With guidance from @lesv and @Neenu1995, our plan is to introduce changes that will be no-op (these change should not affect existing projects) while allowing individual repositories to opt-in GraalVM testing.

Summary of the changes proposed:

* **pom.xml** - Adds an additional Maven profile (named `native`) which contains the dependencies and plugins to run all unit and integration tests compiled with GraalVM with a single command (`mvn test -Pnative -Penable-integration-tests`). This should also be a transparent change for existing projects as the changes are only enabled with activation of the profile.

A demo of this performed on the Secret Manager repo: https://github.com/googleapis/java-secretmanager/pull/583

cc/ @suztomo

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -778,5 +778,78 @@
...
+      </build>
+
+      <repositories>
+        <repository>
+          <id>snapshots-repo</id>
+          <url>https://google.oss.sonatype.org/content/repositories/snapshots</url>
+          <releases><enabled>false</enabled></releases>
+          <snapshots><enabled>true</enabled></snapshots>
+        </repository>
+      </repositories>
```
Comment (lesv) [inline pom.xml:851]: is this needed?

Comment (lesv) [reply pom.xml:851]: Can't you just use the -Pallow-snapshots

**Review by lesv — APPROVED** [review]
approve w/ question

**Comment by Neenu1995** [comment]
Waiting to change the artifact name before merging.

File: pom.xml (hunk 1 above)
Comment (dzou) [reply pom.xml]: Done, I removed this. We will no longer be using snapshots as we will try to eventually pursue putting configuration into java-core or one of the shared dependencies.

**Comment by dzou** [comment]
> Waiting to change the artifact name before merging.

Artifact renamed to `native-image-support` and updated in this commit: https://github.com/googleapis/java-shared-config/pull/314/commits/7926bd4f3e71b9feb012e7ba9708dc9b7e100d22

**Pull request merged**